lib/dotcom_web/controllers/schedule/green.ex

```diff
@@ -75,25 +69,19 @@ defmodule DotcomWeb.ScheduleController.Green do
 
   def predictions(conn, _opts) do
     {predictions, vehicle_predictions} =
-      if DotcomWeb.ScheduleController.Predictions.should_fetch_predictions?(conn) do
-        predictions_stream =
-          conn
-          |> conn_with_branches
-          |> Task.async_stream(
-            fn conn ->
-              DotcomWeb.ScheduleController.Predictions.predictions(conn)
-            end,
-            timeout: @task_timeout,
-            on_timeout: :kill_task
-          )
-
-        vehicle_predictions =
-          DotcomWeb.ScheduleController.Predictions.vehicle_predictions(conn)
-
-        {flat_map_results(predictions_stream), vehicle_predictions}
-      else
-        {[], []}
-      end
+      conn
+      |> conn_with_branches
+      |> Task.async_stream(
+        fn branch_conn ->
+          call_plug(branch_conn, DotcomWeb.ScheduleController.Predictions)
```

Comment:
The `DotcomWeb.ScheduleController.Predictions.should_fetch_predictions?` logic is executed in this plug itself, so it's not needed to be run again in this module.